### PR TITLE
Dockerize application

### LINKED ISSUE #6: Dockerize application
After you're done with #5, dockerize application.
 - Create a Dockerfile for your application
 - Create docker-compose.yml file that will run mongodb and your dockerized application
   - map volumes well, so the nodemon will be restarting your application inside container as you'll be editing the code

### PR DESCRIPTION
**Pull request by jpitucha**
resolves #6

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+FROM node:14.15.1
+
+RUN mkdir -p /usr/src/app
+WORKDIR /usr/src/app
```
Comment (michaelKurowski) [inline Dockerfile:4]: maybe /usr/app instead, you already have src dir in your app

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+COPY ["package.json", "/usr/src/app"]
+
+RUN npm i
+
+RUN npm install -g nodemon
+
+COPY . .
```
Comment (michaelKurowski) [inline Dockerfile]: Why is that

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
+FROM node:14.15.1
+
+RUN mkdir -p /usr/src/app
+WORKDIR /usr/src/app
+
+COPY ["package.json", "/usr/src/app"]
+
+RUN npm i
+
+RUN npm install -g nodemon
```
Comment (michaelKurowski) [inline Dockerfile]: Move nodemon to dev deps, so you won't need to have it installed globally.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+COPY . .
+
+CMD [ "npm", "run", "start" ]
```
Comment (michaelKurowski) [inline Dockerfile]: Just `npm start` :)
Some scripts don't require the `run` keyword

**Review by michaelKurowski — CHANGES_REQUESTED** [review]
Of course you also need to implement docker-compose.yml
But for now I've did a mini-cr for u

File: Dockerfile (hunk 4 above)
Comment (michaelKurowski) [reply Dockerfile]: I think actually that only `npm run start` has a shorthand

**Comment by michaelKurowski** [comment]
You can base on this:
https://github.com/Jordan141/articleblog/blob/master/docker-compose.yml

1. mongo section can stay the same. 
2. You need to change volumes. Basically, map src from here to src in docker
3. I don't think the command part is needed here
4. Ports are a mapping of container ports to your localhost ones


If you'll do it correctly, `docker-compose up` should setup the environment cleanly. You can at first do it without volume mapping, it would make it work without hot reloading, but at least you'd have a feeling that it works at this point and then proceed to volume mapping.

File: Dockerfile (hunk 2 above)
Comment (jpitucha) [reply Dockerfile]: it copyies source files to docker

File: Dockerfile (hunk 3 above)
Comment (jpitucha) [reply Dockerfile]: yea, i forgot remove this line. it is already in deps

**Review by Jordan141 — COMMENTED** [review]
Progress is a good, continue on with docker-compose

**Review by michaelKurowski — APPROVED** [review]

**Comment by michaelKurowski** [comment]
I pass my approval, if everything is done, and @Jordan141 also approves, you guys can merge this branch.

**Review by Jordan141 — APPROVED** [review]
Good work!

**Pull request merged**